File: solutions/dart/prime-factors/1/lib/prime_factors.dart
class PrimeFactors {
  List<int> factors(int number) {
    var factors = <int>[];
    var candidate = 2;

    while (candidate * candidate <= number) {
      if (number % candidate == 0) {
        factors.add(candidate);
        number ~/= candidate;
      } else {
        candidate++;
      }
    }

    if (number > 1) {
      factors.add(number);
    }

    return factors;
  }
}

File: solutions/python/prime-factors/1/prime_factors.py
def factors(value):
    res=[]
    n=2
    while n*n <=value:
        if value%n==0:
            res.append(n)
            value=int(value/n)
        else:
            n+=1

    if value>1:
        res.append(value)
    
    return res

